pkg/docker/config/config.go

```diff
@@ -649,30 +649,43 @@ func findAuthentication(ref reference.Named, registry, path string, legacyFormat
 		keys = []string{registry}
 	}
 
+	// It may be possible that auth.json has been externally modified to
+	// contain an entry like "https://quay.io/v1/". In that case we still want
+	// to match.
+	strippedAuths := map[string]dockerAuthConfig{}
+	for k, v := range auths.AuthConfigs {
+		strippedAuths[stripRegistry(k)] = v
+	}
+
 	// Repo or namespace keys are only supported as exact matches. For registry
 	// keys we prefer exact matches as well.
 	for _, key := range keys {
 		if val, exists := auths.AuthConfigs[key]; exists {
 			return decodeDockerAuth(val)
 		}
+		if val, exists := strippedAuths[key]; exists {
```

Comment:
The comment above talks about exact matches, while this adds a non-exact match.

I’d expect only the “bad luck” code below to need changing.

Reply:
Removed that code part.